File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/BlueSlaver_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class BlueSlaver_Model extends MonsterInfo{
    public static Id = "BlueSlaver"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(48, 52);
        else
            return Mathf.RandomRange(46, 50);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/blueSlaver/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/blueSlaver/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/blueSlaver/skeleton.png"));
    }

    get Name(): string {
        return "蓝色奴隶贩子";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/BlueSlaver_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class BlueSlaver_Model : MonsterInfo
    {
        public static string Id = "BlueSlaver";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(48, 52) : Random.Range(46, 50);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/blueSlaver/skeleton.atlas",
                    "Monsters/theBottom/blueSlaver/skeleton.json", "Monsters/theBottom/blueSlaver/skeleton.png");
            }
        }

        public override string Name => "蓝色奴隶贩子";

        public BlueSlaver_Model(Vector2 pos) : base(pos)
        {
        }
    }
}